coalib/parsing/SimplerConfParser.py

```diff
@@ -0,0 +1,72 @@
+import os
+import configparser
+from collections import OrderedDict
+
+from coalib.misc import Constants
+from coalib.settings.Section import Section
+from coalib.settings.Setting import Setting
+
+
+class SimplerConfParser:
+
+    def __init__(self):
+        self.__init_sections()
+
+    def parse(self, input_data, overwrite=False):
+        """
+        Parses the input and adds the new data to the existing.
+
+        :param input_data: The filename to parse from.
+        :param overwrite:  If True, wipes all existing Settings inside this
+                           instance and adds only the newly parsed ones. If
+                           False, adds the newly parsed data to the existing
+                           one (and overwrites already existing keys with the
+                           newly parsed values).
+        :return:           A dictionary with (lowercase) section names as keys
+                           and their Setting objects as values.
+        """
+        if os.path.isdir(input_data):
+            input_data = os.path.join(input_data, Constants.default_coafile)
+
+        if overwrite:
+            self.__init_sections()
+
+        raw = configparser.ConfigParser(comment_prefixes=None,
+                                        allow_no_value=True)
+        with open(input_data, 'r', encoding='utf-8') as _file:
+            raw.read_file(_file)
+
+        for name in raw.sections():
+            section = self.sections.setdefault(
+                name.lower(), Section(name, self.sections['default']))
+            for keys, value in raw.items(name):
+                if keys.startswith('#'):
+                    self.__add_comment(section, comment=keys, origin=input_data)
+                else:
+                    comments = []
+                    lines, value = value.split('\n'), ''
+                    for line in lines:
+                        if ' #' in line:
```

Comment:
@sils @Udayan12167 Or better: It is actually already handled implicitly by comment search. Now even better with `re.match(r'[^\\]#', ...)` check.